### ISSUE TITLE
Java Configuration Support

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-security-kerberos#7]

**Comment by jvalkeal** [comment]
Should we target this beyond 1.0.0.RELEASE order to get through milestone/rc asap? Only new features I'm currently adding is a better test support and RestTemplate #9.

**Comment by RobMaskell** [comment]
Could we at least document an example java config as of the 1.0.0 release, I have one that works now but would have saved a world of pain if I'd started with an example.

**Comment by jvalkeal** [comment]
Yes, I'm planning to add samples which uses javaconfig and I can add those instructions to docs. I think what @rwinch meant was a specific javaconfig classes so that you'd have same type of functionality compared to spring security where you can just use adapter classes and you don't really need to add any `@Bean`s directly to `@Configuration` classes.

**Comment by rwinch** [comment]
@jvalkeal Yes creating the Java Configuration can wait till after a 1.0
